common/src/main/java/me/lucko/luckperms/common/calculator/LuckyPerms.java

```diff
@@ -0,0 +1,145 @@
+    /** Is it an unlucky day? */
+    private static final boolean UNLUCKY_DAY;
+    /** The unlucky number. */
+    private static final int UNLUCKY_NUMBER = 666;
+    /** The lucky number */
+    private static final int LUCKY_NUMBER;
+
+    static {
+        LocalDate date = LocalDate.now();
+        UNLUCKY_DAY = date.getDayOfWeek() == DayOfWeek.FRIDAY && date.getDayOfMonth() == 13;
+
+        if (Locale.getDefault().getLanguage().equals(Locale.CHINESE.getLanguage())) {
+            // https://en.wikipedia.org/wiki/Chinese_numerology#Eight
+            LUCKY_NUMBER = 8;
+        } else {
+            // https://en.wikipedia.org/wiki/7
+            LUCKY_NUMBER = 7;
+        }
+    }
+
+    /** A ~magical~ counter used when determining luckiness. */
+    private final AtomicInteger counter;
+
+    public LuckyPerms() {
+        this.counter = new AtomicInteger();
+    }
+
+    /**
+     * Uses highly advanced, patented LuckPerms(tm) calculations to determine
+     * the current luckiness measure.
+     *
+     * <p>Returns {@link Tristate#TRUE} if lucky, {@link Tristate#FALSE} if unlucky,
+     * and {@link Tristate#UNDEFINED} if unable to determine.</p>
+     *
+     * @return the calculated luckiness measure
+     */
+    public Tristate calculateLuckiness() {
+        int i = this.counter.incrementAndGet();
+
+        // is the counter divisible by the unlucky number? oh no.. that's not a good sign!
+        if (i % UNLUCKY_NUMBER == 0) {
+            return Tristate.FALSE;
+        }
+
+        // got lucky on a dice roll & the counter is in the lucky numbers set? must be lucky!
+        if (isEven(rollDice()) && LUCKY_NUMBERS.contains(i % 1000)) {
+            return Tristate.TRUE;
+        }
+
+        // is it Friday 13th? I sure hope not.
+        if (UNLUCKY_DAY) {
+            return Tristate.FALSE;
+        }
+
+        // is the current system time divisible by the lucky number? sounds promising?!
+        if ((System.currentTimeMillis() / 1000) % LUCKY_NUMBER == 0) {
+            return Tristate.TRUE;
+        }
+
+        // well, this is embarrassing.
+        return Tristate.UNDEFINED;
+    }
+
+    // Roll a 6-sided fair dice.
+    private static int rollDice() {
+        return ThreadLocalRandom.current().nextInt(6) + 1;
+    }
+
+    // Is the number even?
+    private static boolean isEven(int i) {
+        return i % 2 == 0;
+    }
+
+    // Generates 'lucky numbers'
+    // see: https://en.wikipedia.org/wiki/Lucky_number
+    private static Set<Integer> generateLuckyNumbers(int n) {
+        // generate a sequence 1..n
+        List<Integer> sequence = new ArrayList<>(n);
+        for (int i = 1; i <= n; i++) {
+            sequence.add(i);
+        }
+
+        int j = 1; // the index of the value being considered as the next step
+        int step = 2; // the step to use when deleting
+
+        // delete sequence[j]th elements in the list
+        while (step < sequence.size()) {
+            int idx = step - 1;
```

Comment:
Good day Mr. Luck,

I am the Chief Executive Director of Nitpicking here at the KryptonMC Organisation. I have carefully scanned your work for scrutiny, and I am utterly appalled. As such, I hereby declare that the following variable name "`idx`" is unsatisfactory, and request its immediate removal and replacement with a more fitting variable name.

I have kindly compiled a list of possible replacements for this variable name for you to take inspiration from:
- `a`
- `i`
- `fourtyTwo`
- `magic`

Kindest Regards,
Callum Seabrook
Chief Executive Director of Nitpicking at KryptonMC